improver/calculate_sleet_prob.py

```diff
@@ -57,9 +57,8 @@ def calculate_sleet_probability(prob_of_snow, prob_of_rain):
         ValueError: If the cube contains negative values for the the
                     probability of sleet.
     """
-    ones = np.ones((prob_of_snow.shape), dtype="float32")
-    sleet_prob = ones - (prob_of_snow.data + prob_of_rain.data)
-    if np.any(sleet_prob < 0.0):
+    sleet_prob = 1 - (prob_of_snow.data + prob_of_rain.data)
```

Comment:
I don't think there's any reason to cast the ones here to float32 as `int - float32 = float32` (it's also about twice as fast to do it this way)

Reply:
Chris is right. By casting the ones as float32, the code was forcing the output to be at least float32 precision. Now it is determined by Numpy based on the highest precision of the snow and rain inputs. Other changes in this PR mean they will be int8, so the output here is int8 too. If they are float32 fractions, then the output here would be a float32 fraction too.

Reply:
Ah OK, I misunderstood.  I thought the data coming in were float32 probabilities.

Could you update the docstring please to clarify that these are actually integer / boolean masks, not probabilities?  Otherwise this is fine.